src/app/shared/services/file-manager/file-manager.service.ts

```diff
@@ -1,43 +1,146 @@
+      // To save to user's general Download folder, requires saving to to the "ExternalStorage" directory, with subdirectory "Download".
+      //However this requires additional permissions on Android versions <= 10, which are considered "dangerous"
+      directory = "Documents",
+      subdirectory,
+    } = options;
+    await this.templateAssetService.ready();
+    const blob = (await this.templateAssetService.fetchAsset(relativePath, "blob")) as Blob;
+    try {
+      if (Capacitor.isNativePlatform()) {
+        const { localFilepath } = await this.saveFile({
+          data: blob,
+          targetPath: relativePath,
+          directory,
+          subdirectory,
+        });
+        if (open) FileOpener.open({ filePath: localFilepath, openWithDefault: false });
+      } else {
+        const filename = relativePath.split("/").pop();
+        saveAs(blob, filename);
+      }
+    } catch (err) {
+      this.errorHandler.handleError(err);
+    }
+  }
+
+  /**
+   * Open an asset file externally.
+   * On native devices, download the file to external storage and prompt to open with native apps.
+   * On web, open the asset file in a new browser tab (or prompt download if non-viewable)
+   * @param relativePath
+   */
+  private async openTemplateAsset(relativePath: string) {
+    if (Capacitor.isNativePlatform()) {
+      // Save to "Cache" directory, which may be deleted in cases of low memory. See https://capacitorjs.com/docs/apis/filesystem#directory
+      this.downloadTemplateAsset({
+        relativePath,
+        open: true,
+        directory: "Cache",
+        subdirectory: "",
+      });
+    } else {
+      await this.templateAssetService.ready();
+      const fileUrl = this.templateAssetService.getTranslatedAssetPath(relativePath);
+      window.open(fileUrl, "_blank");
+    }
+  }
+
+  /**
+   * Save a file to the local filesystem (native only),
+   * @param options.directory the name of the directory in which to save the file.
+   * E.g. the permenent "Data" directory (default) or the temporary "Cache" (see https://capacitorjs.com/docs/apis/filesystem#directory)
+   * @param options.subdirectory Additional folder path to be added between "directory" and target path. The app deployment name is always added.
+   * @returns the local filesystem path to the saved file, in both "file://*" format and usable src format
+   */
+  async saveFile(options: {
+    data: Blob;
+    targetPath: string;
+    directory?: keyof typeof Directory;
+    subdirectory?: string;
+  }) {
+    const { data, targetPath, directory = "Data", subdirectory = "" } = options;
+    const path = (subdirectory ? subdirectory + "/" : "") + `${this.cacheName}/${targetPath}`;
     // Docs for write_blob are here: https://github.com/diachedelic/capacitor-blob-writer#readme
-    const src = await write_blob({
-      directory: Directory.Data,
-      path: `${this.cacheName}/${relativePath}`,
-      blob,
+    const localFilepath = await write_blob({
```

Comment:
nit(non-blocking)   
I can see in the blob-writer docs that duplicate named files will be overwritten. Is this a method we should consider exposing to authors in the future along with some UI popup/confirm/prompt if desirable? e.g. `handle_overwrite : 'overwrite' (default) / 'prompt'`